backend/node/genvm/base.py

```diff
@@ -64,43 +66,57 @@ def _get_contract_class_name(contract_code: str) -> str:
             raise Exception("No class name found")
         return matches[0]
 
-    def _generate_receipt(self, class_name, encoded_object, method_name, args):
-        receipt = {
-            # You can't get the name of the inherited class here
-            "class": class_name,
-            "method": method_name,
-            "args": args,
-            "gas_used": self.contract_runner.gas_used,
-            "mode": self.contract_runner.mode,
-            "contract_state": encoded_object,
-            "node_config": self.contract_runner.node_config,
-            "eq_outputs": self.contract_runner.eq_outputs,
-        }
-
-        return receipt
+    def _generate_receipt(
+        self,
+        class_name: str,
+        encoded_object: str,
+        method_name: str,
+        args: str,
+        execution_result: ExecutionResultStatus,
+        error: Exception,
+    ) -> Receipt:
+        return Receipt(
+            class_name,
+            method_name,
+            args,
+            self.contract_runner.gas_used,
+            self.contract_runner.mode,
+            encoded_object,
+            self.contract_runner.node_config,
+            self.contract_runner.eq_outputs,
+            execution_result,
+            error,
```

Comment:
nit: I'd set them through their keywords in order to not depend on the order of the class fields

```suggestion
        return Receipt(
            class_name=class_name,
            method=method_name,
            args=args,
            gas_used=self.contract_runner.gas_used,
            mode=self.contract_runner.mode,
            contract_state=encoded_object,
            node_config=self.contract_runner.node_config,
            eq_outputs=self.contract_runner.eq_outputs,
            execution_result=execution_result,
            error=error,
```

Reply:
Done!